Source/Storages/Realm/Extensions/RealmContext+StorageContext.swift

```diff
@@ -79,7 +79,7 @@ extension RealmContext {
             
             entities.lazy
                 .flatMap { return $0 as? Object }
-                .forEach { realm.add($0, update: false) }
+                .forEach { realm.add($0, update: true) }
```

Comment:
Thank you for pointing it out.

Looking at the documentation, if we enable it and the entity doesn't have a primary key, Realm throws an error. We can see in the Realm implementation:

```
public func add(_ object: Object, update: Bool = false) {
    if update && object.objectSchema.primaryKeyProperty == nil {
        throwRealmException("'\(object.objectSchema.className)' does not have a primary key and can not be updated")
    }
    RLMAddObjectToRealm(object, rlmRealm, update)
}
```
Maybe, it's better if we leave `false` (omitting the parameter) and update the entities with the `update` method? Do you have a case where you need to update with the `add` method? If you need it, we can add an abstract check if the entity has a primary key to avoid throwing the error.

cc @ennioma

Reply:
I just thought that If Realm used that flag it could be exposed, but I understand your point.
My case is: I need to add an entity, then in some future iterations I need to replace that entity with a new one (coming from the network). Without the update flag I should delete the entity (if exists) and add the new one.

Reply:
Hi @sirioz,
you could do something like

```
try context.update {
    oldTask.field01 = newTask.field01
    oldTask.field02 = newTask.field02
    ...
}
```

Isn't it acceptable?

Reply:
You case makes perfectly sense.

Then, I think that we can change like this:

```
let canUpdateIfExists = $0.objectSchema.primaryKeyProperty != nil
realm.add($0, update: canUpdateIfExists)
```

Does it make sense to you? Feel free to use a better approach.

Reply:
@ennioma updating each field could not read so good.
In both cases, updating fields or deleting the old entity and adding a new one, I'd have to make a fetch to retrieve the old record. Not so bad, but it would add code and complexity where it could be simply solved with 1 line `context.add()`.
Now that said, I think that the @MarcoSantarossa's proposal could be a very smart solution.

Reply:
I deeply thought about this topic during the night: the semantic of "add" is of "adding" new things, not updating the old ones. So if you want leave it all as it is now, it's working fine for me.